src/pip/_internal/metadata/base.py

```diff
@@ -359,6 +363,17 @@ def metadata(self) -> email.message.Message:
         """
         raise NotImplementedError()
 
+    @property
+    def json_metadata(self) -> Dict[str, Any]:
+        """PEP 566 compliant JSON-serializable representation of METADATA or PKG-INFO.
+
+        This should return an empty dict if the metadata file is unavailable.
```

Comment:
Is there a reason behind this design choice? My instinct would be to raise an error instead.